x/evm/keeper/grpc_query.go

```diff
@@ -254,8 +256,17 @@ func (k Keeper) EthCall(c context.Context, req *types.EthCallRequest) (*types.Ms
 	return res, nil
 }
 
+const (
+	FromTypeRPC      = 1
+	FromTypeInternal = 2
+)
+
 // EstimateGas implements eth_estimateGas rpc api.
 func (k Keeper) EstimateGas(c context.Context, req *types.EthCallRequest) (*types.EstimateGasResponse, error) {
+	return k.EstimateGasInternal(c, req, FromTypeRPC)
+}
+
+func (k Keeper) EstimateGasInternal(c context.Context, req *types.EthCallRequest, fromType int) (*types.EstimateGasResponse, error) {
```

Comment:
Please, add comment to explain why we need this function
and refactor to use `CallType` instead for an `int`

Reply:
done, please help to check the comment. It maybe not the best.